tests/ui/redundant_pattern_matching_option.rs

```diff
@@ -103,3 +104,34 @@ fn issue7921() {
     if let None = *(&None::<()>) {}
     if let None = *&None::<()> {}
 }
+
+fn issue10726() {
+    let x = Some(42);
+    let y = None::<()>;
+
+    match x {
+        Some(_) => true,
+        _ => false,
+    };
+
+    match x {
+        None => true,
+        _ => false,
+    };
+
+    match y {
+        Some(_) => true,
+        _ => false,
+    };
+
+    match y {
+        None => true,
+        _ => false,
+    };
```

Comment:
These match statements are still equivalent to the ones above

```suggestion
    match x {
        Some(_) => false,
        _ => true,
    };

    match x {
        None => false,
        _ => true,
    };
```

Using y doesn't have any benefit here.